skillmap/src/App.tsx

```diff
@@ -222,68 +222,57 @@ class AppImpl extends React.Component<AppProps, AppState> {
 
             let currentUser = await getUserStateAsync();
             let headerIds = getFlattenedHeaderIds(localUser, state.pageSourceUrl, currentUser);
-
             // Tell the editor to transfer local skillmap projects to the cloud.
             const headerMap = (await res.sendMessageAsync!({
                 type: "pxteditor",
                 action: "savelocalprojectstocloud",
                 headerIds
             } as pxt.editor.EditorMessageSaveLocalProjectsToCloud)).resp.headerIdMap;
-
             if (headerMap) {
-                headerIds = headerIds.map(h => headerMap[h] || h);
-
-                // Patch all of the header ids in the user state and copy
-                // over the local progress that doesn't exist in the signed in
-                // user
-                const urls = Object.keys(currentUser.mapProgress);
                 const newUser: UserState = {
                     ...currentUser,
                     mapProgress: {}
                 }
 
-                for (const url of urls) {
+                // Patch all of the header ids in the user state and copy
+                // over the local progress that doesn't exist in the signed in
+                // user
+                const localUrls = Object.keys(localUser.mapProgress);
+                for (const url of localUrls) {
                     newUser.mapProgress[url] = {
-                        ...currentUser.mapProgress[url],
-                    };
-
-                    if (!localUser.mapProgress[url]) continue;
+                        ...localUser.mapProgress[url]
```

Comment:
shouldn't this be currentuser?

Reply:
my logic is that we copy the local progress, and overwrite with the cloud one at the end if it exists?

Reply:
oh, can you add a comment explaining that?